ui/raidboss/popup-text.ts

```diff
@@ -640,160 +633,28 @@ export class PopupText {
-            const defaultRegex = trigger.regex;
-            const localeRegex = triggerObject[regexParserLang];
-            if (localeRegex instanceof RegExp) {
-              trigger.localRegex = Regexes.parse(localeRegex);
-              orderedTriggers.push(trigger);
-              found = true;
-            } else if (defaultRegex) {
-              const trans = translateRegex(defaultRegex, this.parserLang, set.timelineReplace);
-              trigger.localRegex = Regexes.parse(trans);
-              orderedTriggers.push(trigger);
-              found = true;
-            }
-          }
-
-          // `netRegexDe` (etc) is also deprecated, but they also may still be used by
-          // user triggers.  If they exist, they will take precedence over `netRegex`.
-          // `netRegex` will be auto-translated into the parser language.  `netRegexDe`
-          // and friends will never be auto-translated and are assumed to be correct.
-          if (isNetRegexTrigger(trigger)) {
-            const defaultNetRegex = trigger.netRegex;
-            const localeNetRegex = triggerObject[netRegexParserLang];
-            if (localeNetRegex instanceof RegExp) {
-              // localized regex don't need to handle net-regex auto build
-              trigger.localNetRegex = Regexes.parse(localeNetRegex);
-              orderedTriggers.push(trigger);
-              found = true;
-            } else if (defaultNetRegex) {
-              // simple netRegex trigger, need to build netRegex and translate
-              if (defaultNetRegex instanceof RegExp) {
-                const trans = translateRegex(defaultNetRegex, this.parserLang, set.timelineReplace);
-                trigger.localNetRegex = Regexes.parse(trans);
-                orderedTriggers.push(trigger);
-                found = true;
-              } else {
-                if (trigger.type === undefined) {
-                  console.error(`Trigger ${id}: without type property need RegExp as netRegex`);
-                  continue;
-                }
-
-                const re = buildNetRegexForTrigger(
-                  trigger.type,
-                  translateRegexBuildParam(defaultNetRegex, this.parserLang, set.timelineReplace),
-                );
-                trigger.localNetRegex = Regexes.parse(re);
-                orderedTriggers.push(trigger);
-                found = true;
-              }
-            }
-          }
-
-          if (!found) {
-            console.error('Trigger ' + trigger.id + ': missing regex and netRegex');
-            continue;
-          }
+      if (set.timelineTriggers) {
+        for (const trigger of set.timelineTriggers) {
+          // Timeline triggers are never translated.
+          timelineTriggers.push(this.ProcessTimelineTrigger(trigger));
         }
       }
 
       if (set.overrideTimelineFile) {
+        // TODO: set.filename won't be undefined
         const filename = set.filename ? `'${set.filename}'` : '(user file)';
```

Comment:
How can `set.filename` be undefined now?

```suggestion
        const filename = set.filename;
```